src/fabric/src/fabric2_fdb.erl

```diff
@@ -240,16 +240,39 @@ open(#{} = Db0, Options) ->
     load_validate_doc_funs(Db3).
 
 
+refresh(#{tx := undefined, name := DbName, md_version := OldVer} = Db) ->
+    case fabric2_server:fetch(DbName) of
+        % Relying on these assumptions about the `md_version` value:
+        %  - It is bumped every time `db_version` is bumped
+        %  - Is a versionstamp, so we can check which one is newer
+        %  - If it is `not_found`, it would sort less than a binary value
+        #{md_version := Ver} = Db1 when Ver > OldVer ->
+            Db1#{
+                db_options := maps:get(db_options, Db),
```

Comment:
Seems odd to overwrite the db_options here wholesale given that we've asserted Db1 is newer. Which options would be different?

Reply:
`user_ctx` I think. Which makes me think we should fix why that needs to be in the options and not just as a member in the `Db` map.

Reply:
Good catch. We strip user_ctx out of the options anyway in create and open:

https://github.com/apache/couchdb/blob/prototype/fdb-layer/src/fabric/src/fabric2_fdb.erl#L172-L173

And we set user ctx separately here, and you're right we shouldn't overwrite the newer options with possibly a stale one.